test/WebSites/HtmlGenerationWebSite/Views/HtmlGeneration_Home/Index.cshtml

```diff
@@ -19,7 +19,7 @@
         <a asp-action="Index">Default Controller</a>
     </div>
     <div>
-        <a asp-fragment="fragment" asp-controller="Product">Product Index Fragment</a>
+        <a asp-fragment="fragment" asp-controller="HtmlGeneration_Product">Product Index Fragment</a>
```

Comment:
Controller is actually named - `HtmlGeneration_ProductController`. Since `ProductController` doexn't exist, it runs in to the same issue as https://github.com/aspnet/Routing/issues/728.